test/integration/sagemaker/test_default_inference.py

```diff
@@ -0,0 +1,114 @@
+
+
+@pytest.mark.cpu_test
+def test_default_inference_cpu(sagemaker_session, image_uri, instance_type):
+    instance_type = instance_type or "ml.c4.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(sagemaker_session, image_uri, instance_type, default_model_tar, default_model_script)
+
+
+@pytest.mark.gpu_test
+def test_default_inference_gpu(sagemaker_session, image_uri, instance_type):
+    instance_type = instance_type or "ml.p2.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(sagemaker_session, image_uri, instance_type, default_model_tar, default_model_script)
+
+
+@pytest.mark.skip(reason="Latest EIA version - 1.5.1 uses mms. Enable when EIA images use torchserve")
+@pytest.mark.eia_test
+def test_default_inference_eia(sagemaker_session, image_uri, instance_type, accelerator_type):
+    instance_type = instance_type or "ml.c4.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(
+        sagemaker_session,
+        image_uri,
+        instance_type,
+        default_model_eia_tar,
+        default_model_eia_script,
+        accelerator_type=accelerator_type,
+    )
+
+
+def _test_default_inference(
+    sagemaker_session, image_uri, instance_type, model_tar, mnist_script, accelerator_type=None
+):
+    endpoint_name = sagemaker.utils.unique_name_from_base("sagemaker-pytorch-serving")
+
+    model_data = sagemaker_session.upload_data(
+        path=model_tar,
+        key_prefix="sagemaker-pytorch-serving/models",
+    )
+
+    pytorch = PyTorchModel(
+        model_data=model_data,
+        role="SageMakerRole",
+        predictor_cls=RealTimePredictor if not accelerator_type else PyTorchPredictor,
+        entry_point=mnist_script,
+        image=image_uri,
+        sagemaker_session=sagemaker_session,
+    )
+    with timeout_and_delete_endpoint(endpoint_name, sagemaker_session, minutes=30):
+        # Use accelerator type to differentiate EI vs. CPU and GPU. Don't use processor value
+        if accelerator_type is not None:
+            predictor = pytorch.deploy(
+                initial_instance_count=1,
+                instance_type=instance_type,
+                accelerator_type=accelerator_type,
+                endpoint_name=endpoint_name,
+            )
+        else:
+            predictor = pytorch.deploy(
+                initial_instance_count=1, instance_type=instance_type, endpoint_name=endpoint_name
+            )
+
+        if accelerator_type:
+            batch_size = 100
+            data = np.random.rand(batch_size, 1, 28, 28).astype(np.float32)
+            output = predictor.predict(data)
+            assert output.shape == (batch_size, 10)
```

Comment:
Why is the data and assert different for EIA vs CPU/GPU? Aren't they both the same or similar models?

Reply:
And why is the output of `predictor.predict(...)` handled differently in the two cases? In one case, `.shape` is looked up directly on it, while the other case decodes it and passes it to `json.loads`. What's going on here?

Reply:
Ok...looking further in the review, I found my answer. EIA path uses mnist with default input/output handlers, while CPU/GPU path uses resnet18 with transform_fn defined. Got it!

Reply:
Since I haven't been able to test EIA images with the new toolkit, and the test is skipped by default, I copied over the test prediction request from the `test_mnist.py` test. The success or failure of this EIA test will only be known once we start running it on an image that is capable of using the new toolkit.